core/trino-main/src/main/java/io/trino/eventlistener/EventListenerManager.java

```diff
@@ -144,7 +150,13 @@ private static Map<String, String> loadEventListenerProperties(File configFile)
     public void queryCompleted(Function<Boolean, QueryCompletedEvent> queryCompletedEventProvider)
     {
         try (TimeStat.BlockTimer _ = queryCompletedTime.time()) {
+            if (concurrentQueryCompletedEvents.incrementAndGet() > maxConcurrentQueryCompletedEvents) {
```

Comment:
The limit is being applied globally, so if you have multiple listeners and only one of the is slow, it will cause events to be missed for the other listeners.